cpp/src/groupby/sort/group_quantiles.cu

```diff
@@ -68,11 +69,13 @@ struct calculate_quantile_fn {
     thrust::for_each_n(thrust::seq,
                        thrust::make_counting_iterator(0),
                        num_quantiles,
-                       [d_result = d_result, segment_size, offset](size_type j) {
-                         if (segment_size == 0)
+                       [d_result = d_result, segment_size, offset, this](size_type j) {
+                         if (segment_size == 0) {
                            d_result.set_null(offset + j);
-                         else
+                           atomicAdd(this->null_count, 1);
```

Comment:
Generally I would avoid atomics and compute this count in a separate kernel that works on the bitmask. I vaguely remember seeing anecdotal performance results on a similar type of thing, and I think the benchmarks favored a separate kernel instead of relying on atomics if there is any sizeable number of nulls. The case of all-null columns is too common to disregard here, in my opinion, and would have poor performance with every thread hammering the atomics.

Reply:
I agree that the alternative is likely faster for a meaningful number of nulls, but would prefer to see real benchmarks on how many nulls are required before making any changes here. This isn't the only place where we have this pattern. If we're going to change something we should scope this out more broadly.

Reply:
I'm fine with that approach, if we're doing this same thing elsewhere. I haven't inventoried the libcudf code base to see other places where this is being done.